NewsFeed/Common/Views/ScrollableSegmentedControl.swift

```diff
@@ -0,0 +1,228 @@
+//
+//  ScrollableSegmentedControl.swift
+//  ScrollableSegmentedControl
+//
+//  Created by MArko Satlan on 22/07/2018.
+//  Copyright © 2018 MArko Satlan. All rights reserved.
+//
+
+import UIKit
+
+class ScrollableSegmentedControl: UIControl {
+
+// MARK: - Properties
+    // Open
+    open var font = UIFont(name: "Georgia", size: 15) {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var buttonSpacing: CGFloat = 15 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var buttonMargin: CGFloat = 15 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+    open var selectedSegmentIndex: Int = -1 {
+        didSet {
+            highlightSelectedSegment()
+            calculateAndSetScrollOffset()
+      
+            // deselect previous button if different button is tapped
+            if oldValue != selectedSegmentIndex {
+                let button = buttonsArray.filter{$0.tag == oldValue}.first
+                if let previousButton = button {
+                    previousButton.isSelected = false
+                    UIButton.animate(withDuration: 0.3, animations: {
+                        previousButton.transform = CGAffineTransform.identity
+                    })
+                }
+            }
+        }
+    }
+    
+    // Private
+    private var buttonTitles: [String] = []
+    private var buttonsArray: [UIButton] = []
+    private let scrollView = UIScrollView()
+    
+// MARK: - Methods
+    // Init
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        
+        setUp()
+        drawSeparatorLine()
+    }
+    
+    public init(items: [String]) {
+        self.buttonTitles = items
+        
+        super.init(frame: CGRect.zero)
+        
+        setUp()
+        createButtons()
+        drawSeparatorLine()
```

Comment:
`drawSeparatorLine()` is now repeated in both `init` methods. You can move it to `setUp()` as well.